### PR TITLE
Add overloads of MolOps::getMolFrags and MolOps::getMolFragsWithQuery

### PR DESCRIPTION
**Pull request by DavidACosgrove**
#### Reference Issue
Addresses my question in #7822

#### What does this implement/fix? Explain your changes.
Despite what the branch name might suggest, this just adds a new overload of getMolFrags and of getMolFragsWithQuery that fill in a std::vector<std::unique_ptr<ROMol>> for people like me who are trying to get away from boost::shared_ptr.

#### Any other comments?
I attempted a v2 of the API that did away with boost::shared_ptr<ROMol> but crashed and burned.  This is the minimum solution that does what I need for now.

### CODE REVIEW

File: Code/GraphMol/MolOps.cpp [hunk 1]
```diff
@@ -660,21 +660,21 @@ std::vector<std::unique_ptr<MolSanitizeException>> detectChemistryProblems(
   return res;
 }
 
-std::vector<ROMOL_SPTR> getMolFrags(const ROMol &mol, bool sanitizeFrags,
-                                    INT_VECT *frags,
-                                    VECT_INT_VECT *fragsMolAtomMapping,
-                                    bool copyConformers) {
+namespace {
+std::vector<ROMol *> getRawFrags(const ROMol &mol, bool sanitizeFrags,
```
Comment (greglandrum) [inline Code/GraphMol/MolOps.cpp:664]: A problem I see here is that we're going to leak memory if anything goes wrong in `getRawFrags()`

I think it's safer to have a base function that returns a vector of unique_ptrs and then have the function that needs to return shared_ptrs construct those from the unique_ptrs.

**Comment by greglandrum** [comment]
> I attempted a v2 of the API that did away with boost::shared_ptr but crashed and burned. This is the minimum solution that does what I need for now.

This does seem like the idea case for the v2 API. What problems did you encounter?

File: Code/GraphMol/MolOps.cpp (hunk 1 above)
Comment (DavidACosgrove) [reply Code/GraphMol/MolOps.cpp:664]: I did wonder/worry about that situation.  However, the original code creates a bare molecule and then wraps it in a smart pointer so some of those concerns already existed.  I guess there's no reason why we can't make the code better along the way, though, so I'll do as you suggest.

Comment (greglandrum) [reply Code/GraphMol/MolOps.cpp:664]: > I did wonder/worry about that situation. However, the original code creates a bare molecule and then wraps it in a smart pointer so some of those concerns already existed. I guess there's no reason why we can't make the code better along the way, though, so I'll do as you suggest.

The difference is that the original code creates the new molecule and then immediately wraps it in a shared ptr. Unless I missed something (which is entirely possible), I think there are a bunch more cases where the new code can leak

**Comment by DavidACosgrove** [comment]
My basic problem with a v2 api was not really understanding how the model worked.  Working by analogy with the SmilesParse code, I duplicated the function declarations in the include file inside the v2 namespace, wrapped the originals in a v1 namespace and started implementing the v1 functions inline by calling the corresponding v2 functions.  I hit the buffers trying to create a v1::Hybridization object from a v2 one.  There didn't seem to be a relevant constructor or any obvious way of making one and I didn't know what it was trying to achieve.  At which point I took the path of least resistance, rolled it all back and did it like in this PR which solved my immediate problem with less yak-shaving.  And of course that's how technical debt builds up, so I could be persuaded to try again with someone to point me in the right direction.  Although bit of a bummer that I threw all the code away in the meantime...

There are only these 2 functions that need to be any different between v1 and v2, if all we're caring about is shared_ptr vs unique_ptr, but as far as I could tell the whole lot need duplicating as above.

At the same time as doing this, I thought I could at least future-proof the Python wrappers by using the new api with std::unique_ptr but there didn't seem at present to be a means for boost to delete the molecules via that, and when I had a look at how the file parsers worked, I saw that the helper functions turn std::unique_ptr<ROMol> into boost::shared_ptr<ROMol> to return them.  I concluded that there's a way to go before getting rid of the dependencies on shared pointers completely.

File: Code/GraphMol/MolOps.cpp (hunk 1 above)
Comment (DavidACosgrove) [reply Code/GraphMol/MolOps.cpp]: No, I definitely made things worse from that perspective.  All tightened up now, I think.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**